pymodbus/transport/transport.py

```diff
@@ -177,36 +181,32 @@ def __init__(
             host = self.comm_params.host
             port = int(self.comm_params.port)
         if self.comm_params.comm_type == CommType.SERIAL:
-            host, port = self.init_setup_serial(host, port)
-            if not host and not port:
+            if NULLMODEM_HOST in host:
+                host, port = NULLMODEM_HOST, int(host[9:].split(":")[1])
+            elif self.is_server and host.startswith("socket"):
+                # format is "socket://<host>:port"
+                self.comm_params.comm_type = CommType.TCP
+                ## FIXME: does this make sense??  # pylint: disable=fixme
```

Comment:
This doesn't make sense to me.  Why would `self.comm_params.comm_type == CommType.Serial` and then be changed to `CommType.TCP`?!

Reply:
https://github.com/pymodbus-dev/pymodbus/pull/1710/files#diff-232ea64eea656684ba1d20bc100847b5fabad3d4627d4b8a51c49041ad84897dR30-R31

```
Pyserial allow the comm_port to be a socket e.g. "socket://localhost:502",
this allows serial clients to connect to a tcp server with RTU framer.
```

Reply:
Yes for clients not for servers.